src/extra/AutoUI/Filters/FocusSearch.tsx

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { Txt } from "../../../components/Txt";
+import { Checkbox } from "../../../components/Checkbox";
+import reject from "lodash/reject";
+import groupBy from "lodash/groupBy";
+import styled from "styled-components";
+import { Box } from "../../../components/Box";
+import { Flex } from "../../../components/Flex";
+import { useHistory } from "../../../hooks/useHistory";
+import { AutoUIContext } from "../schemaOps";
+import { stopEvent } from "../../../utils";
+import { AutoUIModel } from "..";
+
+const Focus = styled(Box)`
+	flex-basis: 500px;
+	background-color: white;
+	border: solid 1px ${(props) => props.theme.colors.quartenary.dark};
+	max-height: 200px;
+	position: absolute;
+	width: 100%;
+	z-index: 1;
+	border-radius: 0 0 ${(props) => props.theme.global.drop.border.radius}
+		${(props) => props.theme.global.drop.border.radius};
+	overflow: hidden;
+`;
+
+const FocusContent = styled(Box)`
+	max-height: 180px;
+	overflow-y: auto;
+	overflow-x: auto;
+`;
+
+const FocusItem = styled(Box)<{ hasGetBaseUrl: boolean }>`
+	cursor: ${(props) => (props.hasGetBaseUrl ? "pointer" : "default")};
+	&:hover {
+		background: #dde1f0; // This is the background color Select uses for entities on hover. We do not have it in our theme
+	}
+`;
+
+interface FocusSearchProps<T extends { id: number; [key: string]: any }> {
+	searchTerm: string;
+	filtered: T[];
+	selected: T[];
+	setSelected: (selected: T[]) => void;
+	autouiContext: AutoUIContext<T>;
+	model: AutoUIModel<T>;
+	hasUpdateActions?: boolean;
+}
+
+export const FocusSearch = <T extends { id: number; [key: string]: any }>({
+	searchTerm,
+	filtered,
+	selected,
+	setSelected,
+	autouiContext,
+	model,
+	hasUpdateActions,
+}: FocusSearchProps<T>) => {
+	const history = useHistory();
+
+	const queryTerms = searchTerm.split(" ").map((x) => x.toLowerCase());
+
+	const filteredById = groupBy(
+		filtered.map((entity) => ({
+			id: entity.id,
+			searchTerms: Object.values(entity).filter((val) => val?.length > 0),
```

Comment:
by doing this you are keeping all values that are strings or arrays, skipping numbers. 
Probably better to keep only strings and numbers or only strings.

In case you want to also keep numbers then searchTerms: (string | number)[]

Reply:
I just add `typeof val === 'number' ||` before the `val?.length > 0`. Refer to the above comment for why I think `searchTerms: (string | number)[]` won't work